mrmustard/lab_dev/circuit_components.py

```diff
@@ -233,6 +233,37 @@ def __matmul__(self, other: CircuitComponent) -> CircuitComponent:
 
         return CircuitComponent.from_attributes("", wires_ret, representation_ret)
 
+    def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
+        r"""
+        Contracts ``self`` and ``other`` as it would in a circuit, adding the adjoints when
+        they are missing.
+        """
+        ret = self @ other
```

Comment:
make it more readable by removing the ret pls. 
raise an error in these two cases: self.bra with other.ket and vice versa.

Reply:
Thanks. Done, let me know if it's still hard to parse